packages/backend/src/modules/finality/analyzers/StarknetFinalityAnalyzer.ts

```diff
@@ -0,0 +1,80 @@
+import { assert } from '@l2beat/backend-tools'
+import {
+  ProjectId,
+  TrackedTxsConfigSubtype,
+  UnixTime,
+} from '@l2beat/shared-pure'
+import { BigNumber, utils } from 'ethers'
+import { z } from 'zod'
+
+import { RpcClient } from '../../../peripherals/rpcclient/RpcClient'
+import { StarknetClient } from '../../../peripherals/starknet/StarknetClient'
+import { LivenessRepository } from '../../tracked-txs/modules/liveness/repositories/LivenessRepository'
+import { BaseAnalyzer } from './types/BaseAnalyzer'
+
+const ZBigNumber = z.instanceof(BigNumber).transform(Number)
```

Comment:
Please use Bigint and transform to number safely inside `extractBlockNumber` function
```suggestion
const ZBigNumber = z.instanceof(BigNumber).transform(Number)
```

Reply:
https://github.com/l2beat/l2beat/pull/3258/commits/3d5d874d1694462980e715896a10871ec9171e23